proposals/NNNN-dxil18.md

```diff
@@ -0,0 +1,262 @@
+├───────────────────────────────┼────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
+│getRemainingRecursionLevels    │  240   │call i32 @dx.op.getRemainingRecursionLevels(i32)                                                                       │
+└───────────────────────────────┴────────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+```
+
+### New DXIL Metadata
+
+Node shader entry metadata can contain the following tagged entries:
+
+```
+┌─────────────────────────────────────────┬──────────┬──────────────────────────┐
+│                   Tag                   │ Constant │          Value           │
+├─────────────────────────────────────────┼──────────┼──────────────────────────┤
+│kDxilNumThreadsTag                       │    4     │MDList: (i32, i32, i32)   │
+├─────────────────────────────────────────┼──────────┼──────────────────────────┤
+│kDxilNodeLaunchTypeTag                   │    13    │MDList: (i32)             │
+├─────────────────────────────────────────┼──────────┼──────────────────────────┤
+│kDxilNodeIsProgramEntryTag               │    14    │MDList: (i1)              │
+├─────────────────────────────────────────┼──────────┼──────────────────────────┤
+│kDxilNodeIdTag                           │    15    │MDList: (MDString, i32)   │
+├─────────────────────────────────────────┼──────────┼──────────────────────────┤
+│kDxilNodeLocalRootArgumentsTableIndexTag │    16    │MDList: (i32)             │
+├─────────────────────────────────────────┼──────────┼──────────────────────────┤
+│kDxilShareInputOfTag                     │    17    │MDList: (MDString, i32)   │
+├─────────────────────────────────────────┼──────────┼──────────────────────────┤
+│kDxilNodeDispatchGridTag                 │    18    │MDList: (i32, i32, i32)   │
+├─────────────────────────────────────────┼──────────┼──────────────────────────┤
+│kDxilNodeMaxRecursionDepthTag            │    19    │MDList: (i32)             │
+├─────────────────────────────────────────┼──────────┼──────────────────────────┤
+│kDxilNodeInputsTag                       │    20    │MDList: (MDList[])        │
+├─────────────────────────────────────────┼──────────┼──────────────────────────┤
+│kDxilNodeOutputsTag                      │    21    │MDList: (MDList[])        │
+├─────────────────────────────────────────┼──────────┼──────────────────────────┤
+│kDxilNodeMaxDispatchGridTag              │    22    │MD list: (i32, i32, i32)  │
+├─────────────────────────────────────────┼──────────┼──────────────────────────┤
+│kDxilRangedWaveSize                      │    23    │MD list: (i32, i32, i32)  │
+└─────────────────────────────────────────┴──────────┴──────────────────────────┘

```

Comment:
The Value column doesn't look right.  Each value seems to imply that it points to another MDList containing a value or values.  This isn't the case.  The single values here are operands in-line after the associated tag that identifies the next operand.  The multiple values *are* another MDList, containing that set of values.

The node inputs/outputs is confusing, as it appears to me like an MDList containing a single MDList operand with some set of entries (which might imply that this set of entries are the inputs or outputs).

The meaning/encoding of most of the metadata values is not described.

```suggestion
┌─────────────────────────────────────────┬──────────┬──────────────────────────────┐
│                   Tag                   │ Constant │          Value               │
├─────────────────────────────────────────┼──────────┼──────────────────────────────┤
│kDxilNumThreadsTag                       │    4     │MDList: (i32 x, i32 y, i32 z) │
├─────────────────────────────────────────┼──────────┼──────────────────────────────┤
│kDxilNodeLaunchTypeTag                   │    13    │i32                           │
├─────────────────────────────────────────┼──────────┼──────────────────────────────┤
│kDxilNodeIsProgramEntryTag               │    14    │i1                            │
├─────────────────────────────────────────┼──────────┼──────────────────────────────┤
│kDxilNodeIdTag                           │    15    │MDList: (MDString, i32)       │
├─────────────────────────────────────────┼──────────┼──────────────────────────────┤
│kDxilNodeLocalRootArgumentsTableIndexTag │    16    │i32                           │
├─────────────────────────────────────────┼──────────┼──────────────────────────────┤
│kDxilShareInputOfTag                     │    17    │MDList: (MDString, i32)       │
├─────────────────────────────────────────┼──────────┼──────────────────────────────┤
│kDxilNodeDispatchGridTag                 │    18    │MDList: (i32 x, i32 y, i32 z) │
├─────────────────────────────────────────┼──────────┼──────────────────────────────┤
│kDxilNodeMaxRecursionDepthTag            │    19    │i32                           │
├─────────────────────────────────────────┼──────────┼──────────────────────────────┤
│kDxilNodeInputsTag                       │    20    │MDList: (MDList[], ...)       │
├─────────────────────────────────────────┼──────────┼──────────────────────────────┤
│kDxilNodeOutputsTag                      │    21    │MDList: (MDList[], ...)       │
├─────────────────────────────────────────┼──────────┼──────────────────────────────┤
│kDxilNodeMaxDispatchGridTag              │    22    │MDList: (i32 x, i32 y, i32 z) │
├─────────────────────────────────────────┼──────────┼──────────────────────────────┤
│kDxilRangedWaveSize                      │    23    │MDList: (i32 x, i32 y, i32 z) │
└─────────────────────────────────────────┴──────────┴──────────────────────────────┘
```